plugins/basic/python/tk_nuke_basic/plugin_bootstrap.py

```diff
@@ -0,0 +1,179 @@
+    """
+
+    # ---- now we have everything needed to bootstrap. finish initializing the
+    #      manager and logger, authenticate, then bootstrap the engine.
+
+    import sgtk
+
+    # start logging to log file
+    sgtk.LogManager().initialize_base_file_handler("tk-nuke")
+
+    # get a logger for the plugin
+    sgtk_logger = sgtk.LogManager.get_logger("plugin")
+    sgtk_logger.debug("Booting up toolkit plugin.")
+
+    try:
+        # When the user is not yet authenticated, pop up the Shotgun login
+        # dialog to get the user's credentials, otherwise, get the cached user's
+        # credentials.
+        user = sgtk.authentication.ShotgunAuthenticator().get_user()
+    except sgtk.authentication.AuthenticationCancelled:
+        # TODO: show a "Shotgun > Login" menu in nuke
+        sgtk_logger.info("Shotgun login was cancelled by the user.")
+        return
+
+    # Create a boostrap manager for the logged in user with the plug-in
+    # configuration data.
+    toolkit_mgr = sgtk.bootstrap.ToolkitManager(user)
+
+    toolkit_mgr.base_configuration = base_config
+    toolkit_mgr.plugin_id = plugin_id
+
+    # include the bundle cache as a fallback if supplied
+    if bundle_cache:
+        toolkit_mgr.bundle_cache_fallback_paths = [bundle_cache]
+
+    # Retrieve the Shotgun entity type and id when they exist in the
+    # environment. These are passed down through the app launcher when running
+    # in zero config
+    entity = toolkit_mgr.get_entity_from_environment()
+    sgtk_logger.debug("Will launch the engine with entity: %s" % entity)
+
+    # set up a simple progress reporter
+    toolkit_mgr.progress_callback = bootstrap_progress_callback
+
+    # start engine
+    sgtk_logger.info("Bootstrapping the Shotgun engine for Nuke...")
+
+    toolkit_mgr.bootstrap_engine(__get_engine_name(), entity)
+
+    sgtk_logger.debug("Bootstrap complete.")
+
+    # Contrary to other engines, do not clear the SHOTGUN_ENGINE environment variable.
+    # Nuke spawns a new process when doing a File->Open or File->New, which means our
+    # plugin needs to be able to bootstrap a second time.
+
+
+def __get_engine_name():
+    """
+    Retrieves the engine name to use.
+
+    If the plugin is being launched from the SoftwareLauncher, the method will return
+    the value assigned to the ``SHOTGUN_ENGINE`` environment variable. Otherwise
+    the code will detect whether Nuke or Nuke Studio is running and will return
+    ``tk-nuke`` and ``tk-nukestudio`` respectively.
+    """
+    engine_name = os.environ.get("SHOTGUN_ENGINE")
+    if not engine_name:
+        # FIXME: We are assuming that the configuration inside the standalone plugin
+        # has an engine instance with these name. These values should probably come
+        # from the plugin's info.yml file.
+        try:
+            import hiero # noqa
+        except:
+            engine_name = "tk-nuke"
+        else:
+            engine_name = "tk-nukestudio"
```

Comment:
Mostly just a question out of curiosity. Are we specifically not supporting Hiero, or are we just referencing a tk-nukestudio engine instance in the config for both NS and Hiero?